### ISSUE TITLE
Mattermost webhook

### ISSUE THREAD

**Issue by Infernoman** [42wim/matterbridge#206]
Please answer the following questions. 

### Which version of matterbridge are you using?
version: 0.16.0-dev

### Please describe the expected behavior.
Same channel gateways without webhook on mattermost should change the username and profile picture of the user posting.

#### Use logs from running ```matterbridge -debug``` if possible.
Same channel gateways without webhook on mattermost does not currently change the username or profile picture.

#### (be sure to exclude or anonymize private data (tokens/passwords))

```
[irc]
    [irc.freenode]
        Server="irc.freenode.net:6667"
        Nick="CRW-MatterBridge"
        PrefixMessagesWithNick=true
        IgnoreNicks="travis-ci"

[mattermost]
    [mattermost.work]
        useAPI=true
        Server="SERVER"
        Team="crown"
        Login="matterbridge"
        Password="PASSWORD"
        IconURL="https://robohash.org/{NICK}.png?size=48x48"

    [mattermost.general]
        BindAddress="0.0.0.0:9999"
        IconURL="https://robohash.org/{NICK}.png?size=48x48"
        URL="URL"

[slack]
    [slack.hobby]
        useAPI=true
        Token="TOKEN"
        IconURL="https://robohash.org/{NICK}.png?size=48x48"

[discord]
    [discord.game]
        Token="TOKEN"
        WebhookURL="WEBHOOK URL"
        IconURL="https://robohash.org/{NICK}.png?size=48x48"
        Server="Crown"

[general]
RemoteNickFormat="{NICK}<{PROTOCOL}>"



# General Channels
[[gateway]]
name="mygateway"
enable=true
    [[gateway.inout]]
    account="irc.freenode"
    channel="#crowncoin"

    [[gateway.inout]]
    account="mattermost.general"
    channel="town-square"

    [[gateway.inout]]
    account="slack.hobby"
    channel="general"

    [[gateway.inout]]
    account="discord.game"
    channel="ID"





# Slack / Mattermost Gateways
#[[samechannelgateway]]
#   name="samechannelgateway"
#   enable = true
#   accounts = [ "mattermost.work","slack.hobby" ]
#   channels = [ "altcoins","core-dev","core-dev-team","crw-tokens","developer-general","developer-marketing","github-latest","help","random","testing"]
```

**Comment by Infernoman** [comment]
It also looks like profile photos from the mattermost webhook aren't being relayed from one app to the next either.
![hdnympnattil6aafbk2wfa](https://user-images.githubusercontent.com/8021207/27558017-b9e17766-5a89-11e7-981f-231f6dbdd3e0.png)

**Comment by 42wim** [comment]
Only webhooks can override the username, this doesn't work with the API option.

Your other comment about the profiles, the infernoman profile photo looks the same on slack and discord ?
If you're talking about the dex1 and diehard2k9, those aren't photos

**Comment by Infernoman** [comment]
alright the version i was using before was able to over ride the username and profile picture. guess that has been phased out? 
The first post. 4:04PM "~Tron" on mattermost. relayed to slack and discord with the IconURL instead of my actual profile picture as one would assume should happen.

**Comment by 42wim** [comment]
If you are using ```useAPI=true``` you have never been able to override the username and profile picture. (because it is a normal user and normal users aren't able to that).  Only if you use webhooks, it's possible. 

I'm working on a patch so that you can use the "api" and the webhooks together. So that for receiving messages the bot user can be used (so that you don't need a callback to your own server), and for sending messages the webhook is used. Then this will work.

About the profile issue, I looked into the code. I was forgotten that mattermost doesn't support it. I'm adding a note about this.

**Comment by Infernoman** [comment]
so with no way to webhook a private channel on mattermost, it looks like we have to use `useAPI=true` for private channels and the only way to define which user is speaking on a private channel would be to use `PrefixMessagesWithNick=true`?

**Comment by 42wim** [comment]
Correct, this is not a matterbridge limitation but a mattermost one 
```Outgoing webhooks are only supported in public channels.``` 
(https://docs.mattermost.com/developer/webhooks-outgoing.html)

Also don't forget to set ```RemoteNickFormat``` so that the nick is actually there.

**Comment by 42wim** [comment]
@Infernoman Just released v0.16.0-rc1 with the webhook changes I was talking about.

* URL,UseAPI,BindAddress is deprecated. Your config has to be updated.
  * URL => WebhookURL
  * BindAddress => WebhookBindAddress
  * UseAPI => removed
  This change allows you to specify a WebhookURL and a token (slack,discord), so that
  messages will be sent with the webhook, but received via the token (API)
  If you have not specified WebhookURL and WebhookBindAddress the API (login or token)
  will be used automatically. (no need for UseAPI)
